atlasdb-timelock-server/src/main/java/com/palantir/atlasdb/timelock/atomix/InvalidatingLeaderProxy.java

```diff
@@ -74,21 +76,33 @@ protected Object handleInvocation(Object proxy, Method method, Object[] args) th
         } else {
             clearDelegate();
             throw new ServiceUnavailableException(
-                    String.format("This node (%s) is not the leader (%s)", getLocalId(), getLeaderId()),
+                    String.format("This node (%s) is not the leader (%s)", getLocalId(), currentLeaderId),
```

Comment:
OK on the log change. It seems that with incredible luck you might previously have been able to get "This node (X) is not the leader (X)" if an election was called and this node became the leader at just the right moment.